File: scripts/generate_tileset.py
#!/usr/bin/env python3
"""
Script para gerar um tileset placeholder simples
Cria 3 tiles de 32x32: verde (grama), cinza (concreto), marrom (terra)
"""

from PIL import Image

# Configurações
tile_size = 32
num_tiles = 3

# Cores dos tiles
colors = [
    (76, 175, 80),    # Verde (grama)
    (117, 117, 117),  # Cinza (concreto)
    (141, 110, 99),   # Marrom (terra)
]

# Criar imagem
img = Image.new('RGB', (tile_size * num_tiles, tile_size))

# Desenhar cada tile
for i, color in enumerate(colors):
    for x in range(tile_size):
        for y in range(tile_size):
            img.putpixel((i * tile_size + x, y), color)

# Salvar
img.save('public/assets/tileset.png')
print('Tileset criado em public/assets/tileset.png')

File: scripts/generate-tileset.js
// Script simples para gerar um tileset placeholder
// Cria um PNG com 3 tiles de 32x32: verde (grama), cinza (concreto), marrom (terra)

import fs from 'fs';
import { createCanvas } from 'canvas';

const tileSize = 32;
const numTiles = 3;
const canvas = createCanvas(tileSize * numTiles, tileSize);
const ctx = canvas.getContext('2d');

// Tile 0: Verde (grama)
ctx.fillStyle = '#4CAF50';
ctx.fillRect(0, 0, tileSize, tileSize);

// Tile 1: Cinza (concreto)
ctx.fillStyle = '#757575';
ctx.fillRect(tileSize, 0, tileSize, tileSize);

// Tile 2: Marrom (terra)
ctx.fillStyle = '#8D6E63';
ctx.fillRect(tileSize * 2, 0, tileSize, tileSize);

// Salva o arquivo
const buffer = canvas.toBuffer('image/png');
fs.writeFileSync('./public/assets/tileset.png', buffer);
console.log('Tileset criado em public/assets/tileset.png');
